posthog/cdp/templates/tiktok_ads/template_tiktok_pixel.py

```diff
@@ -0,0 +1,309 @@
+    icon_url="/static/services/tiktok.png",
+    category=["Advertisement"],
+    hog="""
+// Adds window.snaptr and lazily loads the TikTok Pixel script
+function initSnippet() {
+    !function (w, d, t) {
+    w.TiktokAnalyticsObject=t;var ttq=w[t]=w[t]||[];ttq.methods=["page","track","identify","instances","debug","on","off","once","ready","alias","group","enableCookie","disableCookie","holdConsent","revokeConsent","grantConsent"],ttq.setAndDefer=function(t,e){t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}};for(var i=0;i<ttq.methods.length;i++)ttq.setAndDefer(ttq,ttq.methods[i]);ttq.instance=function(t){for(
+    var e=ttq._i[t]||[],n=0;n<ttq.methods.length;n++)ttq.setAndDefer(e,ttq.methods[n]);return e},ttq.load=function(e,n){var r="https://analytics.tiktok.com/i18n/pixel/events.js",o=n&&n.partner;ttq._i=ttq._i||{},ttq._i[e]=[],ttq._i[e]._u=r,ttq._t=ttq._t||{},ttq._t[e]=+new Date,ttq._o=ttq._o||{},ttq._o[e]=n||{};n=document.createElement("script")
+    ;n.type="text/javascript",n.async=!0,n.src=r+"?sdkid="+e+"&lib="+t;e=document.getElementsByTagName("script")[0];e.parentNode.insertBefore(n,e)};
+    }(window, document, 'ttq')
+}
+
+export function onLoad({ inputs }) {
+    initSnippet();
+    ttq.load(inputs.pixelId);
+
+    let userProperties = {};
+    for (const [key, value] of Object.entries(inputs.userProperties)) {
+        if (value) {
+            userProperties[key] = value;
+        }
+    };
+    ttq.instance(inputs.pixelId).identify(userProperties);
+}
+export function onEvent({ inputs }) {
+    let eventProperties = {};
+
+    for (const [key, value] of Object.entries(inputs.eventProperties)) {
+        if (value !== undefined) {
+            eventProperties[key] = value;
+        }
+    };
+
+    if (inputs.contentType !== undefined) {
+        eventProperties.content_type = inputs.contentType;
+    }
+
+    ttq.instance(inputs.pixelId).track(
+        inputs.eventType,
+        eventProperties,
+        {
+            event_id: inputs.eventId
+        }
+    );
+}
+""".strip(),
+    inputs_schema=[
+        {
+            "key": "pixelId",
+            "type": "string",
+            "label": "Pixel ID",
+            "description": "You must obtain a Pixel ID to use the TikTok Pixel. If you've already set up a Pixel for your website, we recommend that you use the same Pixel ID for your browser and server events.",
+            "default": "",
+            "secret": False,
+            "required": True,
+        },
+        {
+            "key": "userProperties",
+            "type": "dictionary",
+            "description": "Map of TikTok user parameters and their values. Check out this page for more details: https://business-api.tiktok.com/portal/docs?id=1739585700402178#item-link-Identity%20information%20supported",
+            "label": "User parameters",
+            "default": {
+                "email": "{sha256Hex(lower(person.properties.email))}",
```

Comment:
I'm quite worried about the lack of tests here tbh. We don't get any telemetry atm from these client side pixels so testing is vital and here we have 0 tests.

I think to move forward we can merge but you should follow up with a PR to:
* Move it to Nodejs
* Actually run the compiled JS version using some test harness (this will be tricky but important imo)